File: A3/src/stmt/IfStmt.java
package stmt;
import stack.*;
import dictionary.*;
import exp.Exp;
import model.MyStmtException;
import model.PrgState;

public class IfStmt implements IStmt
{
	Exp exp;
	IStmt thenS;
	IStmt elseS;
	public IfStmt (Exp e, IStmt t, IStmt el)
	{
		exp = e;
		thenS = t;
		elseS = el;
	}
	public String toString()
	{
		return "IF (" + exp.toString() + ") THEN (" + thenS.toString() + ") ELSE (" + elseS.toString() + ")";
	}
	public PrgState execute(PrgState state) throws MyStmtException
	{
		MyIStack<IStmt> stk = state.getExeStack();
		MyIDictionary<String, Integer> symtbl = state.getSymTable();
		if (exp.eval(symtbl) != 0)
		{
			stk.push(thenS);
		}
		else
		{
			stk.push(elseS);
		}
		return state;
	}
}

File: A7/Stmt/IfStmt.cs
using A7.Dictionary;
using A7.Expr;
using A7.Model;
using A7.Stack;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Stmt
{
    class IfStmt: IStmt
    {
        Exp exp;
        IStmt thenS;
        IStmt elseS;
        public IfStmt(Exp e, IStmt t, IStmt el)
        {
            exp = e;
            thenS = t;
            elseS = el;
        }
        public override String ToString()
        {
            return "IF (" + exp.ToString() + ") THEN (" + thenS.ToString() + ") ELSE (" + elseS.ToString() + ")";
        }
        public PrgState Execute(PrgState state)
        {
            MyIStack<IStmt> stk = state.GetExeStack();
            MyIDictionary<String, int> symtbl = state.GetSymTable();
            if (exp.Eval(symtbl) != 0)
            {
                stk.Push(thenS);
            }
            else
            {
                stk.Push(elseS);
            }
            return state;
        }
    }
}
